Source/Private/ARTOSReachability.m

```diff
@@ -44,11 +48,18 @@ - (void)listenForHost:(NSString *)host callback:(void (^)(BOOL))callback {
     _host = host;
     _callback = callback;
 
-    _reachabilityRef = SCNetworkReachabilityCreateWithName(NULL, [host UTF8String]);
+    dispatch_queue_attr_t attrs = dispatch_queue_attr_make_with_qos_class(DISPATCH_QUEUE_SERIAL, QOS_CLASS_UTILITY, DISPATCH_QUEUE_PRIORITY_DEFAULT);
+    _queue = dispatch_queue_create("com.ably.reachability-monitor", attrs);
+
+    _reachabilityRef = CFAutorelease(SCNetworkReachabilityCreateWithName(kCFAllocatorDefault, [host UTF8String]));
```

Comment:
To be honest - It's been a while since I last worked with the manual objC memory management, so apologies if I'm completely wrong here. But - doesn't `CFAutorelease` mean that the object is added to the autoreleasepool, and will be cleared up sometime after the end of the current scope?

Soo... won't the `_reachabilityRef` be nullified at some point, even though it's not our intention?

We're already firing `CFRelease(_reachabilityRef);` in `dealloc`, which makes sense to me.

Reply:
That was my thinking as well and probably the thinking of original author 😄 But apparently core foundation objects needs to be manually released. I have read it in a couple of places, one example is t[his SO answer](https://stackoverflow.com/a/31435846). So I suspect that the problem might have been that this object might have remained in memory pointing to objects that has already been deallocated. But mine is also just a guess. I just wanted to check whether this would make sense to you too.

Reply:
Yeah, I also recall that Core Foundation objects have to be manually released. In this case I think that the `CFRelease(_reachabilityRef)` in the dealloc should work for that purpose, and using `CFAutorelease` here as well might introduce some other issues that we're not expecting. 

Other than that point everything else makes sense to me, so I'll throw in my approval for this PR.

Reply:
Resolving this on behalf of @JakubJankowski as I removed `CFAutoRelease` https://github.com/ably/ably-cocoa/pull/1452/commits/742ffbf4f309850100bc7799ca7ae156c28d8edf